test/e2e/config-schema-check/index.test.ts

```diff
@@ -0,0 +1,58 @@
+import stripAnsi from 'strip-ansi'
+import { createNextDescribe } from 'e2e-utils'
+
+createNextDescribe(
+  'next.config.js schema validating - defaultConfig',
+  {
+    files: {
+      'pages/index.js': `
+    export default function Page() {
+      return <p>hello world</p>
+    }
+    `,
+      'next.config.js': `
+    module.exports = (phase, { defaultConfig }) => {
+      return defaultConfig
+    }
+    `,
+    },
+  },
+  ({ next }) => {
+    it('should validate against defaultConfig', async () => {
+      const output = stripAnsi(next.cliOutput)
+
+      expect(output).not.toContain('Invalid next.config.js options detected')
+    })
+  }
+)
+
+createNextDescribe(
+  'next.config.js schema validating - invalid config',
+  {
+    files: {
+      'pages/index.js': `
+    export default function Page() {
+      return <p>hello world</p>
+    }
+    `,
+      'next.config.js': `
+    module.exports = {
+      experimental: {
+        badKey: 'badValue'
+      }
+    }
+    `,
+    },
+  },
+  ({ next }) => {
+    it('should warn the invalid next config', async () => {
+      await next.fetch('/')
+      const output = stripAnsi(next.cliOutput)
+      const warningTimes = /badKey/.exec(output)?.length
```

Comment:
```suggestion
      const warningTimes = output.match(/badKey/)?.length
```
Should we use `match` here instead?